exemplary_workflow/pyiron/workflow.py

```diff
@@ -0,0 +1,95 @@
+import os
+from pyiron_base import Project
+
+
+# create pyiron project 
+pr = Project("workflow")
+
+# create conda environments for the proprocessing, processing and postprocessing stage 
+for k, v in {
+    "preprocessing": "../source/envs/preprocessing.yaml",
+    "processing": "../source/envs/processing.yaml",
+    "postprocessing": "../source/envs/postprocessing.yaml"
+}.items():
+    pr.conda_environment.create(env_name=k, env_file=v)
+
+
+# Preprocessing
+## generate mesh
+def write_input(input_dict, working_directory):
+    script_name = os.path.join(working_directory, "gmsh.sh")
+    with open(script_name, "w") as f:
+        f.writelines("gmsh -2 -setnumber domain_size " + str(input_dict["domain_size"]) + " unit_square.geo -o square.msh")
+    os.chmod(script_name, 0o744)
+
+gmsh = pr.wrap_executable(
+    job_name="gmsh",
+    executable_str="./gmsh.sh",
+    write_input_funct=write_input,
+    input_dict={"domain_size": 2.0},
+    conda_environment_path=pr.conda_environment.preprocessing,
+    input_file_lst=["../source/unit_square.geo"],
+    execute_job=True,
+)
+
+## convert mesh to xdmf
+meshio = pr.wrap_executable(
+    job_name="meshio",
+    executable_str="meshio convert square.msh square.xdmf",
+    conda_environment_path=pr.conda_environment.preprocessing,
+    input_file_lst=[gmsh.files.square_msh],
+    execute_job=True,
+)
+
+
+# Processing
+## poisson
+def collect_output(working_directory):
```

Comment:
since that function is potentially different in each step (though in that example it is the only step/job that has a non-file output) I would add the name to the function e.g. collect_output_poisson

Reply:
I updated the names of the `collect_output()` and `write_input()` functions to be unique.